widgets/detector_settings.py

```diff
@@ -563,24 +563,46 @@ def __check_if_selected_elements_have_cut_files(self, cuts_list: List,
                 cut_element_str = c.name.split(".")[1]
                 if cut_element_str in current_eff_file.stem:
                     eff_files_elements.append(cut_element_str)
-        for file in eff_files_elements:
-            if current_eff_file.stem not in file:
-                reply = QtWidgets.QMessageBox.warning(
-                    self, "Warning",
-                    f"Selected element {eff_file_path.stem} "
-                    f"do not have a cut file\n"
-                    f"Do you want to continue?",
-                    QtWidgets.QMessageBox.Ok,
-                    QtWidgets.QMessageBox.Cancel)
-                if reply == QtWidgets.QMessageBox.Cancel:
-                    return True
-                elif reply == QtWidgets.QMessageBox.Ok:
-                    self.obj.add_efficiency_file(eff_file_path)
-                    current_eff_files.add(current_eff_file)
-                    return False
-            else:
-                self.obj.add_efficiency_file(eff_file_path)
-                current_eff_files.add(current_eff_file)
+        if len(eff_files_elements) is 0:
```

Comment:
Don't compare numbers by identity (`is`), use equality (`==`). Using `is` is roughly the same as doing `id(x) == id(y)`. CPython caches the first 100 or so number objects, so this works, but it's only an implementation detail. Example of bad things happening:

```python
>>> a = 100
>>> a is 100
<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
True
>>> b = 1000
>>> b is 1000
<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
False
>>> b == 1000
True
```

```suggestion
        if len(eff_files_elements) == 0:
```